include/nbl/builtin/hlsl/blur/common.hlsl

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#ifndef _NBL_BOX_BLUR_COMMONS_
+#define _NBL_BOX_BLUR_COMMONS_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/intrinsics.h>
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+
+//_NBL_STATIC_INLINE_CONSTEXPR uint32_t WORKGROUP_SIZE = 256u;
+//_NBL_STATIC_INLINE_CONSTEXPR uint32_t PASSES_PER_AXIS = 3u;
+//_NBL_STATIC_INLINE_CONSTEXPR uint32_t AXIS_DIM = 3u; // HUH ?
+
+enum WrapMode : uint32_t
+{
+	WRAP_MODE_CLAMP_TO_EDGE,
+	WRAP_MODE_CLAMP_TO_BORDER,
+	WRAP_MODE_REPEAT,
+	WRAP_MODE_MIRROR,
+};
+
+enum BorderColor : uint32_t
+{
+	BORDER_COLOR_FLOAT_TRANSPARENT_BLACK,
+	BORDER_COLOR_INT_TRANSPARENT_BLACK,
+	BORDER_COLOR_FLOAT_OPAQUE_BLACK,
+	BORDER_COLOR_INT_OPAQUE_BLACK,
+	BORDER_COLOR_FLOAT_OPAQUE_WHITE,
+	BORDER_COLOR_INT_OPAQUE_WHITE
+};
+
+struct BoxBlurParams
+{
+	nbl::hlsl::uint32_t4 inputDimensions;
+	nbl::hlsl::uint32_t2 chosenAxis;
+	float radius;
+
+    uint32_t getDirection()
+    {
+		return nbl::hlsl::glsl::bitfieldExtract( inputDimensions.w, 30, 2 );
+    }
+    uint32_t getChannelCount()
+    {
+        return nbl::hlsl::glsl::bitfieldExtract( inputDimensions.w, 27, 3 );
+    }
+
+    uint32_t getWrapMode()
+    {
+        return nbl::hlsl::glsl::bitfieldExtract( inputDimensions.w, 25, 2 );
+    }
+
+    uint32_t getBorderColorType()
+    {
+        return nbl::hlsl::glsl::bitfieldExtract( inputDimensions.w, 22, 3 );
+    }
```

Comment:
please don't use the old GLSL packing, HLSL has bitfields now

Also we can use `uint16_t` and scalar everywhere, so:
- have the input dimensions as a `uint16_t3`
- then the rest except the radius, packed into a sinlge `uint16_t`

Can you remind me why `chosenAxis` wastes 2 full 32bit uints?